lib/vfscore/main.c

```diff
@@ -1283,67 +1283,13 @@ UK_SYSCALL_R_DEFINE(int, rename, const char*, oldpath, const char*, newpath)
 
 	out_error:
 	trace_vfs_rename_err(error);
-	return -error;
+	return ((error > 0) ? -error : error);
```

Comment:
This looks so ugly... I guess something called by `taskat_conv()` returns a positive value that `taskat_conv()` happily passes forward? The `error = ENOENT` could already be changed to a negative value. Btw, I also see the same pattern in `unlikat()`. Again if vfscore is a mess and we know these patches work as is, we can keep it as is given that we're deprecating it anyway.